modules/extension/wmts/src/main/java/org/geotools/ows/wmts/WebMapTileServer.java

```diff
@@ -151,22 +168,37 @@ public WebMapTileServer(WMTSCapabilities capabilities) throws ServiceException,
      * @param capabilities
      * @throws ServiceException
      * @throws IOException
+     * @deprecated Use constructor with serverUrl, capabilities and httpClient.
      */
+    @Deprecated
     public WebMapTileServer(WMTSCapabilities capabilities, HTTPClient httpClient)
             throws ServiceException, IOException {
-        super(capabilities.getRequest().getGetCapabilities().getGet(), httpClient, capabilities);
+        super(extractServerURL(capabilities), httpClient, capabilities);
         setupType();
```

Comment:
How does setuptype react if there is no getCapabilities object, might be best to go with a null and force the code to use the input url as the base, and make sure the type is set to REST

Reply:
I'm sending in the capabilities, so no change there for setupType.
When it comes to the url. The problem is that there are no useful url within the capabilities. What do you mean by input url?

Reply:
In the last resort we can fall back to the URL we got the getCapabilities from, not sure if we do currently,